spec/unit_tests/moab/storage_object_validator_spec.rb

```diff
@@ -15,104 +15,137 @@
-        # v0006
-        expect(verification_array[7]).to eq(Moab::StorageObjectValidator::NO_XML_FILES =>"Version: v0006 Missing all required metadata files")
+      it 'does not call #check_expected_data_sub_dirs because moab does not have the expected version sub dirs' do
+        druid = 'ee000ee0000'
+        druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/ee/000/ee/0000/ee000ee0000'
+        storage_obj = Moab::StorageObject.new(druid, druid_path)
+        storage_obj_validator = described_class.new(storage_obj)
+        expect(storage_obj_validator).not_to receive(:check_expected_data_sub_dirs)
+        storage_obj_validator.validation_errors
       end
-    end
-    context 'under manifest directory' do
-      druid = 'cc000cc0000'
-      druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/cc/000/cc/0000/cc000cc0000'
-      storage_obj = Moab::StorageObject.new(druid, druid_path)
-      storage_obj_validator = described_class.new(storage_obj)
-      verification_array = storage_obj_validator.validation_errors
-      it 'does not have signatureCatalog.xml' do
-        expect(verification_array).to include(Moab::StorageObjectValidator::NO_SIGNATURE_CATALOG => "Version: v0001 Missing signatureCatalog.xml")
+      it 'does not call #check_metadata_dir_file_only because moab does not have the expected data sub dirs' do
+        druid = 'ff000ff0000'
+        druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/ff/000/ff/0000/ff000ff0000'
+        storage_obj = Moab::StorageObject.new(druid, druid_path)
+        storage_obj_validator = described_class.new(storage_obj)
+        expect(storage_obj_validator).not_to receive(:check_metadata_dir_file_only)
+        storage_obj_validator.validation_errors
       end
+    end
+    context 'valid moab' do
+      let(:druid) { 'bj103hs9687' }
+      let(:druid_path) { 'spec/fixtures/good_root01/moab_storage_trunk/bj/103/hs/9687/bj103hs9687' }
+      let(:storage_obj) { Moab::StorageObject.new(druid, druid_path) }
+      let(:storage_obj_validator) { described_class.new(storage_obj) }
+      let(:verification_array) { storage_obj_validator.validation_errors }
 
-      it 'does not have manifestInventory.xml' do
-        expect(verification_array).to include(Moab::StorageObjectValidator::NO_MANIFEST_INVENTORY => "Version: v0002 Missing manifestInventory.xml")
+      it "returns true when moab is in correct format" do
+        expect(verification_array.empty?).to eq true
+      end
+      it 'calls #check_expected_data_sub_dirs if moab has the expected version sub dirs' do
+        expect(storage_obj_validator).to receive(:check_expected_data_sub_dirs).and_return([])
+        storage_obj_validator.validation_errors
+      end
+      it 'calls #check_metadata_dir_file_only if moab has the expected data sub dirs' do
+        expect(storage_obj_validator).to receive(:check_metadata_dir_file_only).and_return([])
+        storage_obj_validator.validation_errors
       end
```

Comment:
add "valid moab" test to show it calls check_manifest_hoo_ha, too?

do you want to add tests to show it calls the version dir checks, too?

Reply:
yup!